dragonflybsd/libs.md

```diff
@@ -0,0 +1,105 @@
+# Instructions for obtaining EFI related libraries on DragonflyBSD
+
+Read [README.md](./README.md) first for general information.
+
+DragonflyBSD already has support for EFI variables in kernel, so no need to use
+a custom one.
+
+The plan is:
+ * prepare environment for building ports
+ * build and install `efivar-bsd` port (might be renamed to `efivar` later)
+ * build a test tool
+ * verify that it works
+
+## Preparing for using ports
+
+`git`, `gcc` and other basic tools should be installed by default, so we only
+need to pull ports tree. Run this and other commands below as a `root` user:
+
+```
+git clone --depth 1 -b efivar https://github.com/3mdeb/DPorts.git /usr/ports
+```
+
+## Building and installing the port
+
+```
+cd /usr/ports/devel/efivar-bsd
+# update "distinfo" file in case it diverged due to active development
+make makesum
+# build and install the port
+make install
+```
+
+## Prepare user-space tool
+
+```
+mkdir ~/efivarlibtest
+cd ~/efivarlibtest
+curl https://raw.githubusercontent.com/3mdeb/openbsd-src/efivars-api/usr.sbin/efivarlibtest/efivarlibtest.c > efivarlibtest.c
+gcc -I/usr/local/include -L/usr/local/lib -Wl,-rpath,/usr/local/lib efivarlibtest.c -lefivar -o efivarlibtest
+```
+
+`-rpath` is needed to avoid picking up `/usr/lib/libefivar.so` which comes with
+the system. Might need to rename port's library to get rid of this.
+
+## Use the test tool
+
+### Adding/updating a variable
+
+```
+./efivarlibtest set aaaaaaaa-bbbb-cccc-dddd-eeeeeeeeeeee test newval
```

Comment:
@SergiiDmytruk I'm getting: `Failed to set var: Function not implemented`

Reply:
Added `kldload efirt` command. I always forget that `/dev/efi` is in a module here.